### DISCUSSION TITLE
[i18n] inconsistent redirect behaviour with defaultLocale and query params

### DISCUSSION THREAD

**Discussion by nclsndr** [vercel/next.js#43380] [Ideas]
**What version of Next.js are you using?**

10.0.7

**What version of Node.js are you using?**

14.15

**What browser are you using?**

Chrome latest

**What operating system are you using?**

macOS

**How are you deploying your application?**

AWS-k8s

**Describe the Bug**

Let's consider I have 2 locales: `['en', 'fr']` where `en` is the defaultLocale.
With the current i18n feat implementation, a page like `about-us`, using `en` locale,  is available both on: `/about-us` and `/en/about-us`.

However, if I try to navigate to `/en/about-us?any=true`, the server does return a 200 with content but once the client is taking the control, we end up having a client rewrite/redirect to `/about-us?any=true`.

**Expected Behavior**

If we access to  `/[defaultLocale]/about-us?any=true` we don't want a (meaningless) client side rewrite/redirect.

**To Reproduce**

Take any Next project using i18n with at least 1 locale defined.
Try to access to any route using defaultLocale and query params. E.g. `localhost:3000/[defaultLocale]?any=true`

**Comment by enzoferey** [comment]
Duplicate of https://github.com/vercel/next.js/issues/21076.

The question is still open though, this only happens when query params are present 🤔

**Comment by jankaifer** [comment]
This is intended behavior (explained in https://github.com/vercel/next.js/issues/21076#issuecomment-760788612).

I will convert this into a discussion because that's where we track feature requests.
